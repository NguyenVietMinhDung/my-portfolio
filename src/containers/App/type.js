export type Props = {
  isVisible: boolean,
  activatedIndex: number,
  activateNavigationItem: Function,
  openNavigationMenu: Function,
};
