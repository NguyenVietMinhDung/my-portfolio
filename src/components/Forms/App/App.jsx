import React from 'react';
import Navbar from '../Navbar';

export default () => (
  <div className="container">
    <Navbar />
  </div>
);
