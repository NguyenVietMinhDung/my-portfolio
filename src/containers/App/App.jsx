// @flow
import React, { memo } from 'react';
import type { Props } from './type';
import constants from '../../constants';
import forms from '../../components/forms';

const {
  header, footer, aboutMe, app, navbar, experiences, educations, projects, achievements,
} = constants;

const {
  Navbar, Header, Footer, AboutMe, Experiences, Educations, Projects, Achievements,
} = forms;

const App = (props: Props) => (
  <>
    <Navbar
      items={navbar.items}
      hamburgerIconUrl={navbar.hamburgerIconUrl}
      closeIconUrl={navbar.closeIconUrl}
      logoUrl={app.logoUrl}
      {...props}
    />
    <Header
      name={header.NAME}
      description={header.DESCRIPTION}
      btnTxt={header.BUTTON_TEXT}
      avatarUrl={header.avatarUrl}
      socialNetworks={app.socialNetworks}
    />
    <AboutMe
      title={aboutMe.TITLE}
      greeting={aboutMe.GREETING}
      summary={aboutMe.SUMMARY}
      basicInfo={aboutMe.basicInfo}
      skills={aboutMe.skills}
    />
    <Experiences
      title={experiences.TITLE}
      timeline={experiences.timeline}
    />
    <Projects
      title={projects.TITLE}
      items={projects.items}
    />
    <Educations
      title={educations.TITLE}
      timeline={educations.timeline}
    />
    <Achievements
      title={achievements.TITLE}
      timeline={achievements.timeline}
    />
    <Footer
      slogan={footer.SLOGAN}
      copyright={footer.COPYRIGHT}
      socialNetworks={app.socialNetworks}
      logoUrl={app.logoUrl}
    />
  </>
);

export default memo<Props>(App);
