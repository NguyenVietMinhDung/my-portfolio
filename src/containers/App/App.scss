.loader {
  position: absolute;
  width: 100vw;
  height: 100vh;
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  align-items: center;
}