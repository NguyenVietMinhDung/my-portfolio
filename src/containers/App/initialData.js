import facebookIconUrl from '../../assets/images/facebook.png';
import instagramIconUrl from '../../assets/images/instagram.png';
import linkedInIconUrl from '../../assets/images/linkedin.png';

export const navbarItems = [
  {
    name: 'Home',
    url: '#',
  },
  {
    name: 'About',
    url: '#',
  },
  {
    name: 'Services',
    url: '#',
  },
  {
    name: 'Portfolio',
    url: '#',
  },
  {
    name: 'Blog',
    url: '#',
  },
  {
    name: 'Contact',
    url: '#',
  },
];

export const basicInfo = [
  {
    iconClassName: 'lni-calendar',
    name: 'Date of birth',
    value: '16 January 1995',
  },
  {
    iconClassName: 'lni-envelope',
    name: 'Email',
    value: 'ngvietminhdung@gmail.com',
  },
  {
    iconClassName: 'lni-phone-handset',
    name: 'Phone',
    value: '(076) 401-2912',
  },
  {
    iconClassName: 'lni-map-marker',
    name: 'Location',
    value: 'Tan Binh District, HCM City',
  },
];

export const socialNetworks = [
  {
    name: 'Facebook',
    url: 'https://www.facebook.com/minh.dung.391',
    iconUrl: facebookIconUrl,
  },
  {
    name: 'Instagram',
    url: 'https://www.instagram.com/dungvmnguyen',
    iconUrl: instagramIconUrl,
  },
  {
    name: 'LinkedIn',
    url: 'https://www.linkedin.com/in/d%C5%A9ng-nguy%E1%BB%85n-502865114',
    iconUrl: linkedInIconUrl,
  },
];

export const skills = [
  {
    itemId: 'skill-1',
    name: 'JavaScript, HTML, CSS',
    value: 100,
  },
  {
    itemId: 'skill-2',
    name: 'ReactJS, Redux',
    value: 80,
  },
  {
    itemId: 'skill-3',
    name: 'SCSS, Styled-Component',
    value: 80,
  },
  {
    itemId: 'skill-4',
    name: 'Jest, Enzyme',
    value: 70,
  },
  {
    itemId: 'skill-5',
    name: 'Webpack',
    value: 80,
  },
  {
    itemId: 'skill-6',
    name: 'Node.js',
    value: 50,
  },
  {
    itemId: 'skill-7',
    name: 'Git',
    value: 80,
  },
];
