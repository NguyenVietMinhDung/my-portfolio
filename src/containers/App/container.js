import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { activateNavigationItem, openNavigationMenu } from '../../actions';
import App from './App';

const mapStateToProps = state => ({
  activatedIndex: state.activatedIndex,
  isVisible: state.isVisible,
});

const mapDispatchToProps = dispatch => bindActionCreators(
  { activateNavigationItem, openNavigationMenu },
  dispatch,
);

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(App);
