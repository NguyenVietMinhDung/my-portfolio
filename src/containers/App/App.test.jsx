import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

const defaultProps = {
  isVisible: false,
  activatedIndex: 0,
  activateNavigationItem: jest.fn(),
  openNavigationMenu: jest.fn(),
};

describe('<App />', () => {
  test('renders correctly', () => {
    const component = shallow(<App {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
});
