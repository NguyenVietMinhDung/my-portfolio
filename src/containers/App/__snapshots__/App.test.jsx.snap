// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<Fragment>
  <Memo(Navbar)
    activateNavigationItem={[MockFunction]}
    activatedIndex={0}
    closeIconUrl="test-file-stub"
    hamburgerIconUrl="test-file-stub"
    isVisible={false}
    items={
      Array [
        Object {
          "itemId": "header",
          "name": "Home",
          "url": "#home",
        },
        Object {
          "itemId": "about-me",
          "name": "About",
          "url": "#about",
        },
        Object {
          "itemId": "work-experiences",
          "name": "Experiences",
          "url": "#experiences",
        },
        Object {
          "itemId": "relevant-projects",
          "name": "Projects",
          "url": "#projects",
        },
        Object {
          "itemId": "my-educations",
          "name": "Educations",
          "url": "#educations",
        },
        Object {
          "itemId": "my-achievements",
          "name": "Achievements",
          "url": "#achievements",
        },
        Object {
          "itemId": "footer",
          "name": "Contact",
          "url": "#contact",
        },
      ]
    }
    logoUrl="test-file-stub"
    openNavigationMenu={[MockFunction]}
  />
  <Memo(Header)
    avatarUrl="test-file-stub"
    btnTxt="DOWNLOAD MY CV"
    description="A Freelance UI Designer & Web Developer"
    name="Dung Nguyen"
    socialNetworks={
      Array [
        Object {
          "iconUrl": "test-file-stub",
          "name": "Facebook",
          "url": "https://www.facebook.com/minh.dung.391",
        },
        Object {
          "iconUrl": "test-file-stub",
          "name": "Instagram",
          "url": "https://www.instagram.com/dungvmnguyen",
        },
        Object {
          "iconUrl": "test-file-stub",
          "name": "LinkedIn",
          "url": "https://www.linkedin.com/in/dungvmn1995",
        },
      ]
    }
  />
  <Memo(AboutMe)
    basicInfo={
      Array [
        Object {
          "iconClassName": "lni-calendar",
          "name": "Date of birth",
          "value": "16 January 1995",
        },
        Object {
          "iconClassName": "lni-envelope",
          "name": "Email",
          "value": "ngvietminhdung@gmail.com",
        },
        Object {
          "iconClassName": "lni-phone-handset",
          "name": "Phone",
          "value": "(076) 401-2912",
        },
        Object {
          "iconClassName": "lni-map-marker",
          "name": "Location",
          "value": "Tan Binh District, HCM City",
        },
      ]
    }
    greeting="Hi There! I'm Dung Nguyen"
    skills={
      Array [
        Object {
          "itemId": "skill-1",
          "name": "JavaScript, HTML, CSS",
          "value": 80,
        },
        Object {
          "itemId": "skill-2",
          "name": "ReactJS",
          "value": 60,
        },
        Object {
          "itemId": "skill-3",
          "name": "SCSS",
          "value": 60,
        },
        Object {
          "itemId": "skill-5",
          "name": "Webpack",
          "value": 50,
        },
        Object {
          "itemId": "skill-6",
          "name": "Node.js",
          "value": 60,
        },
        Object {
          "itemId": "skill-7",
          "name": "Git",
          "value": 60,
        },
      ]
    }
    summary="I am an ambitious Web
Application Developer who
always tries to improve himself
everyday, a quick learner who
can absorb new ideas and can
communicate clearly and
effectively. Looking for an
ambitious company which will
challenge my developer and
allows me to continue
to develop my knowledge and
potential."
    title="About Me"
  />
  <Memo(Experiences)
    timeline={
      Object {
        "iconClassName": "lni-briefcase",
        "items": Array [
          Object {
            "endDate": "May 2020",
            "key": 1,
            "organization": "Ubitec",
            "role": "JAVASCRIPT DEVELOPER",
            "startDate": "September 2019",
          },
          Object {
            "endDate": "August 2019",
            "key": 2,
            "organization": "KMS Technology Inc., Vietnam",
            "role": "SOFTWARE ENGINEER",
            "startDate": "November 2017",
          },
          Object {
            "description": "I had joined KMS as a fresher before graduating. I had been trained to become an official employee for 2 months. Then, I joined one of the biggest project team in the company.",
            "endDate": "November 2017",
            "key": 3,
            "organization": "KMS Technology Inc., Vietnam",
            "role": "FRESHER",
            "startDate": "September 2017",
          },
        ],
      }
    }
    title="Experiences"
  />
  <Memo(Projects)
    items={
      Array [
        Object {
          "description": "A chat tool that helps customers can contact with bankers and versa vice",
          "endDate": "May 2020",
          "isCurrentProject": false,
          "name": "Textme",
          "role": "Developer",
          "startDate": "September 2019",
        },
        Object {
          "description": "A ticket-selling website for US high school events",
          "endDate": "June 2019",
          "isCurrentProject": false,
          "name": "Huddle",
          "role": "Developer",
          "startDate": "January 2019",
        },
        Object {
          "description": "A micro-services project management application
which helps managers to keep track of resource allocation",
          "endDate": "January 2019",
          "isCurrentProject": false,
          "name": "ERTPMTool",
          "role": "Developer",
          "startDate": "July 2018",
        },
        Object {
          "description": "Provider Search is a user interface that allows
users to search for practitioners in the MasterFile. Built on the Single",
          "endDate": "June 2018",
          "isCurrentProject": false,
          "name": "PDV",
          "role": "Developer",
          "startDate": "February 2018",
        },
        Object {
          "description": "An internal tool which supports managers to keep track of resource allocation",
          "endDate": "February 2018",
          "isCurrentProject": false,
          "name": "DMTool",
          "role": "Developer",
          "startDate": "September 2017",
        },
      ]
    }
    title="Relevant Projects"
  />
  <Memo(Educations)
    timeline={
      Object {
        "iconClassName": "lni-graduation",
        "items": Array [
          Object {
            "description": "Computer Science Major",
            "endDate": "April 2018",
            "key": 1,
            "organization": "Bach Khoa University, Ho Chi Minh City",
            "role": "SOFTWARE ENGINEER DEGREE",
            "startDate": "September 2013",
          },
        ],
      }
    }
    title="Educations"
  />
  <Memo(Achievements)
    timeline={
      Object {
        "iconClassName": "lni-certificate",
        "items": Array [
          Object {
            "description": "AWS Certified Cloud Practitioner",
            "endDate": "August 2023",
            "key": 1,
            "organization": "Amazon Web Services (AWS)",
            "startDate": "August 2020",
          },
        ],
      }
    }
    title="Achievements"
  />
  <Memo(Footer)
    copyright="Copyright © 2019 Dung Nguyen • A Freelance UI Designer & Web Developer"
    logoUrl="test-file-stub"
    slogan="Keep It Simple, Stupid"
    socialNetworks={
      Array [
        Object {
          "iconUrl": "test-file-stub",
          "name": "Facebook",
          "url": "https://www.facebook.com/minh.dung.391",
        },
        Object {
          "iconUrl": "test-file-stub",
          "name": "Instagram",
          "url": "https://www.instagram.com/dungvmnguyen",
        },
        Object {
          "iconUrl": "test-file-stub",
          "name": "LinkedIn",
          "url": "https://www.linkedin.com/in/dungvmn1995",
        },
      ]
    }
  />
</Fragment>
`;
